app/controllers/concerns/shared.rb

```diff
@@ -64,6 +68,13 @@ def update_moment(model_object, model_name, moment)
     Moment.find_by(id: moment.id).update(params)
   end
 
+  def update_strategy(model_object, model_name, strategy)
```

Comment:
Not mandatory: update_moment and update_strategy look pretty similar, we could combine them and have some conditional logic for whether to use Moment.find_by or Strategy.find_by